### PR TITLE
edit for vm

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py [hunk 1]
```diff
@@ -38,3 +41,62 @@ def create_container(self, **kwargs):
...
+        vnc = 'vncdotool -s %s' % vnc_ip
+        result,error = self.execute_shell_commands(cmd="%s type %s key enter" % (vnc, repr('ls')))       
+        return error 
+        
+    def enable_ssh_access(self, vnc_ip, username=None, password=None):
+
+        username = username or self.vm_username
+        password = password or self.vm_password
+
+        """
```
Comment (john-kheir) [inline functional_testing/zrobot_templates/framework/zos_utils/utils.py]: Please add the doc string at the beginning of the function

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py [hunk 2]
```diff
@@ -38,3 +41,62 @@ def create_container(self, **kwargs):
...
+
+        response = client.bash(cmd).get()
+        return response
+    
+    def execute_shell_commands(self, cmd):
+        process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, error = process.communicate()
+        return out.decode('utf-8'), error.decode('utf-8')
+
+    def vm_info(self, vm_name):
```
Comment (john-kheir) [inline functional_testing/zrobot_templates/framework/zos_utils/utils.py]: i think it is better to call this function.. get_vm()

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py [hunk 3]
```diff
@@ -38,3 +41,62 @@ def create_container(self, **kwargs):
 
     def create_vm(self, **kwargs):
         return self.handle_blueprint('vm.yaml', **kwargs)
+
+    def check_vnc(self, vnc_ip):
+        vnc = 'vncdotool -s %s' % vnc_ip
+        result,error = self.execute_shell_commands(cmd="%s type %s key enter" % (vnc, repr('ls')))       
+        return error
```
Comment (john-kheir) [inline functional_testing/zrobot_templates/framework/zos_utils/utils.py]: why are u returning error? is there any helpful info other than error?

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py [hunk 4]
```diff
@@ -38,3 +41,62 @@ def create_container(self, **kwargs):
 
     def create_vm(self, **kwargs):
         return self.handle_blueprint('vm.yaml', **kwargs)
+
+    def check_vnc(self, vnc_ip):
+        vnc = 'vncdotool -s %s' % vnc_ip
+        result,error = self.execute_shell_commands(cmd="%s type %s key enter" % (vnc, repr('ls')))       
+        return error 
+        
+    def enable_ssh_access(self, vnc_ip, username=None, password=None):
```
Comment (john-kheir) [inline functional_testing/zrobot_templates/framework/zos_utils/utils.py]: Make sure vnc is installed

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py [hunk 5]
```diff
@@ -38,3 +41,62 @@ def create_container(self, **kwargs):
...
+                self.execute_shell_commands(cmd="%s key shift-_ type config key enter" % vnc)
+                time.sleep(1)
+            elif 'https' in cmd:
+                self.execute_shell_commands(cmd="%s type %s" % (vnc, repr(cmd)))
+                self.execute_shell_commands(cmd="%s key shift-:" % vnc)
+            else:
+                self.execute_shell_commands(cmd="%s type %s key enter" % (vnc, repr(cmd)))
+                time.sleep(1)
+
+    def execute_command_inside_vm(self, client, vmip,  cmd, username=None, password=None):
```
Comment (john-kheir) [inline functional_testing/zrobot_templates/framework/zos_utils/utils.py]: which client?

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py (hunk 2 above)
Comment (Dinaamagdy) [reply functional_testing/zrobot_templates/framework/zos_utils/utils.py]: done

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py (hunk 3 above)
Comment (Dinaamagdy) [reply functional_testing/zrobot_templates/framework/zos_utils/utils.py]: done

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py (hunk 1 above)
Comment (Dinaamagdy) [reply functional_testing/zrobot_templates/framework/zos_utils/utils.py]: done

File: functional_testing/zrobot_templates/framework/zos_utils/utils.py (hunk 5 above)
Comment (Dinaamagdy) [reply functional_testing/zrobot_templates/framework/zos_utils/utils.py]: added doc string  describe parameters

**Pull request merged**